File: Arcade/Intro/13.clj
(defn reverseInParentheses [inputString]
  "Reverse string inside parentheses.
   Solution of luizsol."
  (let [pattern (re-find #"\([^()]*\)" inputString)] ; Use regex to find the right most parentheses
    (if (nil? pattern) inputString
        (reverseInParentheses
         (clojure.string/replace inputString
                                 pattern
                                 (clojure.string/reverse
                                  (subs pattern 1 (dec (count pattern)))))))))

File: Arcade/Intro/13.cpp
#include<regex>

std::string reverseInParentheses(std::string inputString) {
  // Reverse string insides parentheses.
  std::regex inside_par("\\([^()]*\\)");
  std::smatch sm;

  while(std::regex_search(inputString, sm, inside_par)) {
    std::string tmp(sm[0]);
    inputString = std::regex_replace(inputString, inside_par, std::string(tmp.rbegin()+1,tmp.rend()-1) + "$2");
  }
  return inputString;
}
